ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/CreateKeytabFilesServerAction.java

```diff
@@ -217,6 +217,17 @@ protected CommandReport processIdentity(ResolvedKerberosPrincipal resolvedPrinci
                 KerberosPrincipalEntity principalEntity = kerberosPrincipalDAO.find(resolvedPrincipal.getPrincipal());
                 String cachedKeytabPath = (principalEntity == null) ? null : principalEntity.getCachedKeytabPath();
 
+		//This is to hanlde regenerate keytabs for missing hosts if keytab file does not exist in the cache dir.                
+		if(!regenerateKeytabs) {
```

Comment:
@hapylestat If regenerateKeytabs is already true and then don't need to check the file exist - since this operation is already slow - adding "if(!regenerateKeytabs)" check would avoid running file existance check again for keytabs which is not needed.